docs/modules/ROOT/pages/access-control.adoc

```diff
@@ -202,3 +134,143 @@ TIP: A recommended configuration is to grant both roles to a secure governance c
 
 The minimum delay (accessible through the xref:api:governance.adoc#TimelockController-getMinDelay--[`getMinDelay`] method) can be updated by calling the xref:api:governance.adoc#TimelockController-updateDelay-uint256-[`updateDelay`] function. Bear in mind that access to this function is only accessible by the timelock itself, meaning this maintenance operation has to go through the timelock itself.
+
+[[access-management]]
+== Access Management
+
+Although xref:api:access.adoc#AccessControl[`AccessControl`] offers a more dynamic solution for adding permissions to your contracts, decentralized protocols tend to become more complex after integrating new contract instances, requiring keeping track of permissions in each one. This increases the complexity of permissions management and monitoring across the system.
+
+image::access-control-multiple.svg[Access Control multiple]
+
+Protocols managing permissions in production systems often require more integrated alternatives to fragmented permissions through multiple `AccessControl` instances. 
+
+Better role management can be achieved with an xref:api:access.adoc#AccessManager[`AccessManager`] instance. Instead of managing each contract's permission separately, AccessManager stores all the permissions in a single contract, making your protocol easier to audit and maintain.
+
+image::access-manager.svg[Access Manager]
+
+The AccessManager is designed around the concept of role:
+
+* Roles are granted to accounts (addresses) following a many-to-many approach. This means that each user can have one or multiple roles and multiple users can have the same role.
+* Access to a restricted function is restricted to one role. A function is defined by one https://docs.soliditylang.org/en/v0.8.20/abi-spec.html#function-selector[function selector] on one contract (called target). 
+
+For a call to be authorized, the caller must bear the role that is assigned to the current function (contract address + function selector).
+
+image::access-manager-functions.svg[Access Manager functions]
+
+=== Using `AccessManager`
+
+OpenZeppelin Contracts provides xref:api:access.adoc#AccessManager[`AccessManager`]. A contract that can be deployed and used out of the box. It sets an initial admin in the constructor who will be allowed to perform management operations.
+
+In order to restrict access to some functions of your contract, you should inherit from the xref:api:access.adoc#AccessManaged[`AccessManaged`] contract provided along with the manager so that the `restricted` modifier is available to protect any externally facing function. Note you will have to set the address of the Access Manager instance (xref:api:access.adoc#AccessManaged-authority--[authority]) in the constructor so the `restricted` modifier can check permissions by calling the correct manager.
```

Comment:
```suggestion
In order to restrict access to some functions of your contract, you should inherit from the xref:api:access.adoc#AccessManaged[`AccessManaged`] contract provided along with the manager so that the `restricted` modifier is available to protect any externally facing function. Note you will have to set the address of the Access Manager instance (xref:api:access.adoc#AccessManaged-constructor-address-[`initialAuthority`]) in the constructor so the `restricted` modifier can check permissions by calling the correct manager.
```

Reply:
This is a beginner question, but does this mean that the AccessManager needs to be the deployed _before_ the other contracts, so you have an on-chain address you can refer to?

Reply:
Yes, but you can change the authority afterwards. There are some cases where the manager can be deployed afterwards:

- We saw some usage of the [`IAuthority` interface from Dappsys]. Contracts out there can migrate easily from a DS Authority to the Access Manager even if it's deployed afterwards.
- For Ownable and Access Control, the manager can be deployed afterwards and then be granted permissions on them.